experimental/perf/scripts/push_perf_results_helper.py

```diff
@@ -0,0 +1,109 @@
+import argparse
+import os
+import re
+
+import wandb
+
+
+def parse_log(log_path):
+    """Parses a performance test log file and extracts relevant metrics.
+
+    Args:
+        log_path (str): Path to the log file.
+
+    Returns:
+        dict: A dictionary of parsed metrics.
+    """
+    with open(log_path) as f:
+        log_content = f.read()
+
+    # Extract general information
+    steps_per_run = int(
+        re.search(r"# of steps in each run: (\d+)", log_content).group(1)
+    )
+    metrics_per_step = int(
+        re.search(r"# of metrics in each step: (\d+)", log_content).group(1)
+    )
+
+    # Extract timing information for the run
+    init_time = float(re.search(r"init_wandb\(\) time: ([\d.]+)", log_content).group(1))
+    log_metrics_time = float(
+        re.search(r"log_metrics\(\) time: ([\d.]+)", log_content).group(1)
+    )
+    finish_time = float(
+        re.search(r"finish_wandb\(\) time: ([\d.]+)", log_content).group(1)
+    )
+
+    # Extract system metrics
+    system_metrics = {}
+    for match in re.finditer(r"\{([^}]+)\}", log_content):
+        metrics = match.group(1).split(", ")
+        for metric in metrics:
+            key, value = metric.split(":")
+            system_metrics[key] = float(value)
+
+    # Consolidate all extracted data
+    metrics = {
+        "steps_per_run": steps_per_run,
+        "metrics_per_step": metrics_per_step,
+        "timing": {
+            "init_time": init_time,
+            "log_metrics_time": log_metrics_time,
+            "finish_time": finish_time,
+        },
+        "system_metrics": system_metrics,
+    }
+
+    print(metrics)
+    return metrics
+
+
+def log_to_wandb(project_name, root_log_dir, run_name):
+    """Logs the parsed performance test data to W&B.
+
+    Args:
+        project_name (str): Name of the W&B project.
+        log_dir (str): Directory containing performance test logs.
+    """
+    wandb.init(project=project_name, name=run_name, job_type="performance_test")
+
+    print("inside log_to_wandb")
```

Comment:
do you want to keep the print? maybe something like this if so:
```suggestion
    print("[INFO] inside log_to_wandb")
```